x-pack/plugins/ml/public/embeddables/lens/lens_vis_layer_selection_flyout/layer/compatible_layer.tsx

```diff
@@ -0,0 +1,432 @@
+  const [bucketSpan, setBucketSpan] = useState(DEFAULT_BUCKET_SPAN);
+
+  const [jobIdValid, setJobIdValid] = useState<string>('');
+  const [bucketSpanValid, setBucketSpanValid] = useState<string>('');
+  const [state, setState] = useState<STATE>(STATE.DEFAULT);
+  const [createError, setCreateError] = useState<{ text: string; errorText: string } | null>(null);
+
+  function createADJobInWizard() {
+    convertLensToADJob(embeddable, share, layerIndex);
+  }
+
+  async function createADJob() {
+    setState(STATE.SAVING);
+    setCreateError(null);
+    const result = await createAndSaveJob(
+      jobId,
+      bucketSpan,
+      embeddable,
+      startJob,
+      runInRealTime,
+      data.dataViews,
+      kibanaConfig,
+      data.query.timefilter.timefilter,
+      ml,
+      layerIndex
+    );
+    const error = checkForCreationErrors(result);
+    if (error === null) {
+      setState(STATE.SAVE_SUCCESS);
+    } else {
+      setState(STATE.SAVE_FAILED);
+      setCreateError(error);
+    }
+  }
+
+  const viewResults = useCallback(
+    async (jobType: CREATED_BY_LABEL | null) => {
+      const { timeRange } = embeddable.getInput();
+      const page = jobType === CREATED_BY_LABEL.MULTI_METRIC ? 'explorer' : 'timeseriesexplorer';
+      const locator = share.url.locators.get(ML_APP_LOCATOR);
+      if (locator) {
+        const url = await locator!.getUrl({
+          page,
+          pageState: {
+            jobIds: [jobId],
+            timeRange,
+          },
+        });
+
+        application.navigateToUrl(url);
+      }
+    },
+    [jobId, embeddable, share, application]
+  );
+
+  function setStartJobWrapper(start: boolean) {
+    setStartJob(start);
+    setRunInRealTime(start && runInRealTime);
+  }
+
+  useDebounce(
+    async function validateJobId() {
+      setJobIdValid('');
+      setBucketSpanValid('');
+      const validationResults = basicJobValidation(
+        {
+          job_id: jobId,
+          analysis_config: { detectors: [], bucket_span: bucketSpan },
+        } as unknown as estypes.MlJob,
+        undefined,
+        { max_model_memory_limit: '', effective_max_model_memory_limit: '' },
+        true
+      );
+
+      if (validationResults.contains('job_id_invalid')) {
+        setJobIdValid(
+          i18n.translate('xpack.ml.newJob.wizard.validateJob.jobNameAllowedCharactersDescription', {
+            defaultMessage:
+              'Job ID can contain lowercase alphanumeric (a-z and 0-9), hyphens or underscores; ' +
+              'must start and end with an alphanumeric character',
+          })
+        );
+      } else if (validationResults.contains('job_id_invalid_max_length')) {
+        setJobIdValid(
+          i18n.translate('xpack.ml.newJob.wizard.validateJob.jobIdInvalidMaxLengthErrorMessage', {
+            defaultMessage:
+              'Job ID must be no more than {maxLength, plural, one {# character} other {# characters}} long.',
```

Comment:
I have found **_4_** places (excl this PR) in the codebase with this message. Shall we have a single job validator?

Reply:
yes, it would be good to have a single job validator. it's something which has been discussed a few times before. I think that would be outside of the scope of this PR.
Also pulling these strings out of the wizard translations is outside of the scope, but could be a simple follow on task to only have one copy of these strings in the product.
I've reused the translation id here, so to reduce the repetition in the final product.

Reply:
Raised https://github.com/elastic/kibana/issues/139460 to fix this